central/reprocessor/reprocessor_unit_test.go

```diff
@@ -0,0 +1,95 @@
+package reprocessor
+
+import (
+	"testing"
+
+	protobuf "github.com/gogo/protobuf/types"
+	"github.com/golang/mock/gomock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+
+	nodeDatastoreMocks "github.com/stackrox/rox/central/node/datastore/mocks"
+	riskManagerMocks "github.com/stackrox/rox/central/risk/manager/mocks"
+	"github.com/stackrox/rox/generated/storage"
+	nodesEnricherMocks "github.com/stackrox/rox/pkg/nodes/enricher/mocks"
+)
+
+func Test_loopImpl_reprocessNode(t *testing.T) {
+	type args struct {
+		id string
+	}
+	type mocks struct {
+		nodes        *nodeDatastoreMocks.MockDataStore
+		risk         *riskManagerMocks.MockManager
+		nodeEnricher *nodesEnricherMocks.MockNodeEnricher
+	}
+	tests := []struct {
+		name       string
+		args       args
+		node       *storage.Node
+		want       bool
+		setUpMocks func(t *testing.T, a *args, m *mocks)
+	}{
+		{
+			name: "when OS is RHCOS, nothing is done",
+			setUpMocks: func(t *testing.T, a *args, m *mocks) {
+				node := &storage.Node{
+					OsImage: "Red Hat Enterprise Linux CoreOS 412.86.202302091419-0 (Ootpa)",
+				}
+				m.nodes.EXPECT().GetNode(gomock.Any(), a.id).Return(node, true, nil)
+			},
+		},
+		{
+			name: "when OS is not RHCOS, nothing scanner is called and node is upserted",
+			setUpMocks: func(t *testing.T, a *args, m *mocks) {
+				node := &storage.Node{
+					OsImage:     "Something that is not RHCOS",
+					LastUpdated: protobuf.TimestampNow(),
+				}
+				gomock.InOrder(
+					m.nodes.EXPECT().GetNode(gomock.Any(), gomock.Eq(a.id)).Times(1).Return(node, true, nil),
+					m.nodeEnricher.EXPECT().EnrichNode(node).Times(1).Return(nil),
+					m.risk.EXPECT().CalculateRiskAndUpsertNode(gomock.Any()).DoAndReturn(func(node *storage.Node) error {
+						assert.Nil(t, node.LastUpdated, "expected node.LastUpdated = nil when CalculateRiskAndUpsertNode(node) is called")
+						return nil
+					}).Times(1),
+				)
+			},
+			want: true,
+		},
+		{
+			name: "when get node returns err, returns false",
+			setUpMocks: func(t *testing.T, a *args, m *mocks) {
+				m.nodes.EXPECT().GetNode(gomock.Any(), a.id).Times(1).Return(nil, false, errors.New("foobar"))
+			},
+			want: false,
+		},
+		{
+			name: "when get node returns not found, returns false",
+			setUpMocks: func(t *testing.T, a *args, m *mocks) {
+				m.nodes.EXPECT().GetNode(gomock.Any(), a.id).Times(1).Return(nil, false, nil)
+			},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
```

Comment:
Finish is optional here - see https://pkg.go.dev/github.com/golang/mock/gomock#NewController
We do not call finish in `central/sensor/service/pipeline/nodeinventory/pipeline_test.go`, so maybe let's pick one variant (calling it or not calling it) and apply globally.

Reply:
Thanks! Quoting docs:

> New in go1.14+, if you are passing a *testing.T into NewController function you no longer need to call ctrl.Finish() in your test methods. 

Fixed at 751aa60dc2